lib/vector-core/buffers/src/lib.rs

```diff
@@ -75,6 +80,12 @@ where
         } => {
             let (tx, rx) = mpsc::channel(max_events);
             let tx = BufferInputCloner::Memory(tx, when_full);
+            let rx = rx.inspect(|item: &T| {
+                emit(&BufferEventsSent {
+                    count: 1,
+                    byte_size: item.allocated_bytes(),
```

Comment:
Should this be `item.size_of()`? `allocated_bytes` only accounts for the non-struct parts of the size, `size_of` includes all the sizes.

Reply:
Ah, you are right. That was my bad. I had suggested `allocated_bytes`, but you are right that we should call `.size_of()` which the same `ByteSizeOf` implements.

Reply:
Will adjust, I should have clarified. Thanks!